src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
+const DEFAULT_ANIMAL_HEALTH = 100;
+const CARNIVORE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name, health = DEFAULT_ANIMAL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, isHidden = false) {
+    super(name, health);
+    this.hidden = isHidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= CARNIVORE_DAMAGE;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter(animal => animal !== target);
```

Comment:
it's better to create a static method for the class that will delete the animal:

```
if (target.health <= 0) {
  Animal.kill(victim);
}
```